contracts/WildcardSteward_v2.sol

```diff
@@ -466,17 +530,110 @@ contract WildcardSteward_v2 is Initializable {
+            tokenPatron,
+            _newPrice
+        );
+    }
+
+    // Check possible issues with collectPatronage(tokenId) [probably leave out since no patronage to collect on forecolsed token!?]
+    function buyAuction(
+        uint256 tokenId,
+        uint256 _newPrice,
+        uint256 wildcardsPercentage
+    )
+        public
+        payable
+        collectPatronage(tokenId)
+        collectPatronageAddress(msg.sender)
+        priceGreaterThanZero(_newPrice)
+        validWildcardsPercentage(wildcardsPercentage)
+    {
+        require(
+            state[tokenId] == StewardState.Foreclosed,
+            "Can only buy foreclosed tokens useing this function"
+        );
+        // Calculate auction price for the token and replace price[tokenId] below
+        uint256 auctionTokenPrice = _auctionPrice(tokenId);
+
+        // Note: We need to calculate the requisite value on our front end using same function
+        uint256 remainingValueForDeposit = msg.value.sub(auctionTokenPrice);
+
+        address currentOwner = assetToken.ownerOf(tokenId);
+        address tokenPatron = currentPatron[tokenId];
+
+        _buyAuction(tokenId);
+        _handleBuy(
+            tokenId,
+            wildcardsPercentage,
+            remainingValueForDeposit,
+            currentOwner,
+            tokenPatron,
+            _newPrice
+        );
+    }
+
+    function _handleBuy(
+        uint256 tokenId,
+        uint256 wildcardsPercentage,
+        uint256 remainingValueForDeposit,
+        address currentOwner,
+        address tokenPatron,
+        uint256 _newPrice
+    ) internal {
+        wildcardsPercentages[tokenId] = wildcardsPercentage; //Set the percentage wildcards will recieve on next sale
+        deposit[msg.sender] = deposit[msg.sender].add(remainingValueForDeposit);
+        transferAssetTokenTo(
+            tokenId,
+            currentOwner,
+            tokenPatron,
+            msg.sender,
+            _newPrice
+        );
+        emit Buy(tokenId, msg.sender, _newPrice);
+    }
+
+    // NB to check that the msg.value is greater tgat
+    function _buyAuction(uint256 tokenId) internal {
+        uint256 totalAmount = price[tokenId];
+        uint256 artistAmount = totalAmount.mul(100).div(10000); //1% hard coded at the moment
+        uint256 wildcardsAmount = totalAmount.sub(artistAmount);
+
+        _payArtistAndWildcards(tokenId, artistAmount, wildcardsAmount);
+
+        state[tokenId] = StewardState.Owned;
+        timeLastCollected[tokenId] = now;
+        timeLastCollectedPatron[msg.sender] = now;
+    }
 
+    function _buy(
+        uint256 tokenId,
+        address tokenPatron,
+        uint256 wildcardsPercentage
+    ) internal {
+        uint256 totalAmount = price[tokenId];
+        if (wildcardsPercentages[tokenId] == 0) {
+            wildcardsPercentages[tokenId] = 500; // 5% default for existing tokens. Thought this was easier than adding commission rates to existing tokens?
```

Comment:
Yes, this is perfect :) Eventually we won't need this I guess, but Vitalik will probably be the last one to get bought.